util/Migrator/DbScripts_future/2023-10-FutureMigrations.sql

```diff
@@ -0,0 +1,50 @@
+-- Remove old stored procedures and SelectionReadOnlyArray for Flexible Collections
+-- They have been superseded via their respective _V2 variants
```

Comment:
💭 Is there any appetite to then restore the original name? I know we chatted about "v2" and that's certainly a more permanent name we can increment -- and this doesn't need to be applied this round -- but a "vNext" or "vTransition" could make sense if it's then replaced. This would support EDD and keep the codebase cleaner, but that might just be my OCD around ugly version numbers in database changes.

Reply:
I wondered this myself, but then we need a further EDD cycle to change the name back; i.e. there needs to be another transition period where we duplicate `SelectionReadOnlyArray_v2` to the original name (without `_v2`), have both, then delete `v2`. That seems like unnecessary work to me, plus I don't know what happens if another change is required during that time.

Reply:
Yeah, I'm not the biggest fan of leaving the `v2` suffix permanently. But, like @eliykat mentioned, we'd have to introduce another complete cycle to transition the name back and that may be more of a headache than its worth.